### PR TITLE
improve:remove IXRInteraface's dependent from GFXDevice.h

### PR DESCRIPTION
**Pull request by oahc09**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/renderer/gfx-validator/DeviceValidator.h [hunk 1]
```diff
@@ -106,6 +107,7 @@ class CC_DLL DeviceValidator final : public Agent<Device> {
 
     bool _recording{false};
     uint64_t _currentFrame{1U};
+    IXRInterface *_xr{nullptr};
```
Comment (star-e) [inline native/cocos/renderer/gfx-validator/DeviceValidator.h]: Is IXRInterface used in `DeviceValidator`? It seems not used.

Comment (oahc09) [reply native/cocos/renderer/gfx-validator/DeviceValidator.h]: doesn't used, can be removed

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by hana-alice** [comment]
Tested with wgpu(no Agent/Validator), fine with me.

**Review by hana-alice — APPROVED** [review]

**Review by star-e — APPROVED** [review]

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by oahc09** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Pull request merged**